index.js

```diff
@@ -132,53 +162,61 @@ function nowServing(line) {
   if (!line.length) {
     return "There is nobody waiting to be served!"
   }
+  takeATicketNumber(line)
   return `Currently serving ${line.shift()}.`
 }
 
+function takeATicketNumber(line) {
+  line.push(line.length + 1)
+  return `Welcome. You are ticket number ${line.length}` 
+}
 const line = []
 
 // ***** Scope & Closures - Question 1 *****
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 1.`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
+function ticketNumberGeneratorFunc(line){
+  return line 
+}
```

Comment:
It's totally okay if you didn't understand this! We don't expect you to know all of the information you learned today all in one day! :)